clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc

```diff
@@ -91,7 +91,7 @@ Iter3 replace_copy_if(Policy &&policy, Iter1 first, Iter1 last, Iter2 mask,
                        std::random_access_iterator_tag>::value,
       "Iterators passed to algorithms must be random-access iterators.");
   return std::transform(std::forward<Policy>(policy), first, last, mask, result,
-                        internal::replace_if_fun<T, Pred>(p, new_value));
+                        internal::replace_if_fun<typename std::iterator_traits<Iter1>::value_type, Pred>(p, new_value));
```

Comment:
Traits of Iter3 should be used here.

Reply:
Fixed.  SYCLomatic-test will be udpated to hit this corner.